docs/conf.py

```diff
@@ -154,12 +151,25 @@
     "common_links.rst",
 ]
 
-
 # If true, `todo` and `todoList` produce output, else they produce nothing.
 todo_include_todos = False
 
 default_role = "obj"
 
+# Customizations for make linkcheck using regular expressions
+linkcheck_allowed_redirects = {
+    r"https://.+\.(org|io|com)": r".+(org|io|com)/en/.+",  # to different doc branches
+    r"https://doi\.org/.+": r"https://.+",  # DOI links are more persistent
+    r"https://.+": r".+(google|github).+[lL]ogin.+",  # some links require logins
+}
```

Comment:
Since I didn't know this beforehand, `.+` is effectively a wildcard marker, `(...|...)` is either/or, and `[ ]` means matching any one of these characters.